samples/client/petstore/dart-dio/petstore_client_lib/lib/api/pet_api.dart

```diff
@@ -17,504 +17,490 @@ import 'package:openapi/api_util.dart';
+      }..removeWhere((_, dynamic value) => value == null),
+      extra: <String, dynamic>{
+        'secure': <Map<String, String>>[
+          {
+            'type': 'oauth2',
+            'name': 'petstore_auth',
+          },
+        ],
+        ...?extra,
+      },
+      validateStatus: validateStatus,
+      contentType: [
+        'application/json',
+        'application/xml',
+      ].first,
+      cancelToken: cancelToken,
+      onSendProgress: onSendProgress,
+      onReceiveProgress: onReceiveProgress,
+    );
+
+    dynamic _bodyData;
+
+    const _type = FullType(Pet);
+    _bodyData = _serializers.serialize(body, specifiedType: _type);
+
+    final _response = await _dio.request<dynamic>(
+      _request.path,
+      data: _bodyData,
+      options: _request,
+    );
+
+    return _response;
+  }
+
+  /// Deletes a pet
+  ///
+  /// 
+  Future<Response<void>> deletePet(
+    int petId, { 
+    String apiKey,
+    CancelToken cancelToken,
+    Map<String, dynamic> headers,
+    Map<String, dynamic> extra,
+    ValidateStatus validateStatus,
+    ProgressCallback onSendProgress,
+    ProgressCallback onReceiveProgress,
+  }) async {
+    final _request = RequestOptions(
+      path: r'/pet/{petId}'.replaceAll('{' r'petId' '}', petId.toString()),
+      method: 'DELETE',
+      headers: <String, dynamic>{
+        r'api_key': apiKey,
+        ...?headers,
+      }..removeWhere((_, dynamic value) => value == null),
+      queryParameters: <String, dynamic>{
+      }..removeWhere((_, dynamic value) => value == null),
+      extra: <String, dynamic>{
+        'secure': <Map<String, String>>[
+          {
+            'type': 'oauth2',
+            'name': 'petstore_auth',
+          },
+        ],
+        ...?extra,
+      },
+      validateStatus: validateStatus,
+      contentType: [
+        'application/json',
+      ].first,
+      cancelToken: cancelToken,
+      onSendProgress: onSendProgress,
+      onReceiveProgress: onReceiveProgress,
+    );
+
+    dynamic _bodyData;
+
+    final _response = await _dio.request<dynamic>(
+      _request.path,
+      data: _bodyData,
+      options: _request,
+    );
+
+    return _response;
+  }
+
+  /// Finds Pets by status
+  ///
+  /// Multiple status values can be provided with comma separated strings
+  Future<Response<BuiltList<Pet>>> findPetsByStatus(
+    BuiltList<String> status, { 
+    CancelToken cancelToken,
+    Map<String, dynamic> headers,
+    Map<String, dynamic> extra,
+    ValidateStatus validateStatus,
+    ProgressCallback onSendProgress,
+    ProgressCallback onReceiveProgress,
+  }) async {
+    final _request = RequestOptions(
+      path: r'/pet/findByStatus',
+      method: 'GET',
+      headers: <String, dynamic>{
+        ...?headers,
+      }..removeWhere((_, dynamic value) => value == null),
+      queryParameters: <String, dynamic>{
+        r'status': status,
+      }..removeWhere((_, dynamic value) => value == null),
```

Comment:
that looks a lot better

Reply:
I think I will remove this soon and add `nullable` handling to the parameters